docs/proposals/build-skipping.adoc

```diff
@@ -0,0 +1,69 @@
+= Proposal: Multi build repo skipping
+
+== Purpose
+
+Skip builds for which nothing has changed in their working directory.
+
+== Background
+
+For a repo with a single build, the pipeline build will only be triggered if any changes trigger a new pipeline run.
+
+However if one has multiple builds in one repo, the pipeline build runs all build tasks even for those areas where nothing has changed.
+
+A simple scenario is a repo with a backend and frontend component whose build outputs are combined by a subsequent docker build in a subsequent deploy task.
+
+There is no good reason when only the frontend code changes that the backend build is redone as well and vice versa of course.
+
+== Solution
+
+All tasks shall be enabled to support build skipping.
+
+The following is added to the images of these tasks:
+
+- `git`
+
+Task parameter
+
+- `cache-output-dir` a boolean defaulting to `'true'`.
+
+The solution builds on the caching mechanism introduced in issue #147.
+
+`ods-start` supports a new area in the cache for the purpose of caching build task outputs.
+This area is below folder `.ods-cache/build-task`.
+
+If `cache-output-dir` is `'true'` build tasks shall copy their build outputs to `.ods-cache/build-task/$(git-sha-working-dir)`, where
```

Comment:
I am not sure I understand where this path lives and how it relates to the global cache. Also - wouldn't we need further separation below "build task"? Two components in one repo might use the same build task, and might have the same Git SHA. Or is "build task" referring to the name of the task in the pipeline definition?

Reply:
Hmm, I think this situation can happen: imagine a repository which has two Go components. Both components use the same Go task to build outputs, and both may be changed in the same Git commit. In that event, you still would want to separate the output directories ...

Reply:
We had a direct conversation about what the mentioned `$(git-sha-working-dir)` is?
This is the crypto hash of the internal Git tree object which stores the `working-dir` (and all the git managed files and directories beneath it).  
In particular this is not a git commit hash which for example last modified that directory.
(See also https://stackoverflow.com/a/55459589 and https://stackoverflow.com/a/58600111 as well as https://git-scm.com/book/en/v2/Git-Internals-Git-Objects)

While this addressed the original concerns, a build tasks which offers parameters to build variants such as for different platforms would still have a possibility for clashes. 

Perhaps the tasks name in the ods.yaml could be used to avoid a clash here?

An alternative that by now I favor is that the relevant parameters would be used automatically. For example for go these might be derived by using the task parameters for `go-os`, `go-arch` and perhaps `enable-cgo`. 

So one might use folder `.ods-cache/build-task/go-$(go-os)-$(go-arch)-$(enable-go)/$(git-sha-working-dir)`

@michaelsauter let me know if I forgot something here. We could leave this unresolved until we capture this somewhere in the final code/docs or at least in this proposal.